packages/twenty-front/src/modules/auth/hooks/useAuth.ts

```diff
@@ -350,20 +402,33 @@ export const useAuth = () => {
         throw new Error('No login token');
       }
 
+      if (isEmailVerificationRequired) {
+        setSearchParams({ email });
+        setSignInUpStep(SignInUpStep.EmailVerification);
+        return null;
+      }
+
       if (isMultiWorkspaceEnabled) {
         return redirectToWorkspaceDomain(
           signUpResult.data.signUp.workspace.subdomain,
-          AppPath.Verify,
+          isEmailVerificationRequired ? AppPath.SignInUp : AppPath.Verify,
           {
-            loginToken: signUpResult.data.signUp.loginToken.token,
+            ...(!isEmailVerificationRequired && {
+              loginToken: signUpResult.data.signUp.loginToken.token,
+            }),
+            email,
           },
         );
       }
 
-      const { user, workspace, workspaceMember } = await handleVerify(
+      const verifyResult = await handleVerify(
```

Comment:
We will have conflict here.

Now handleVerify return nothing.

Reply:
Resolving as a recent commit fixes this.